application/src/main/java/org/togetherjava/tjbot/commands/reminder/RemindRoutine.java

```diff
@@ -0,0 +1,139 @@
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.*;
+import net.dv8tion.jda.api.requests.RestAction;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.Routine;
+import org.togetherjava.tjbot.db.Database;
+
+import java.awt.*;
+import java.time.Instant;
+import java.time.temporal.TemporalAccessor;
+import java.util.concurrent.TimeUnit;
+
+import static org.togetherjava.tjbot.db.generated.Tables.PENDING_REMINDERS;
+
+/**
+ * Routine that processes and sends pending reminders.
+ * <p>
+ * Reminders can be set by using {@link RemindCommand}.
+ */
+public final class RemindRoutine implements Routine {
+    private static final Logger logger = LoggerFactory.getLogger(RemindRoutine.class);
+    private static final Color AMBIENT_COLOR = Color.decode("#F7F492");
+    private static final int SCHEDULE_INTERVAL_SECONDS = 30;
+    private final Database database;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param database the database that contains the pending reminders to send.
+     */
+    public RemindRoutine(@NotNull Database database) {
+        this.database = database;
+    }
+
+    @Override
+    public @NotNull Schedule createSchedule() {
+        return new Schedule(ScheduleMode.FIXED_RATE, 0, SCHEDULE_INTERVAL_SECONDS,
+                TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void runRoutine(@NotNull JDA jda) {
+        Instant now = Instant.now();
+        database.write(context -> context.selectFrom(PENDING_REMINDERS)
+            .where(PENDING_REMINDERS.REMIND_AT.lessOrEqual(now))
+            .stream()
+            .forEach(pendingReminder -> {
+                sendReminder(jda, pendingReminder.getId(), pendingReminder.getGuildId(),
+                        pendingReminder.getChannelId(), pendingReminder.getAuthorId(),
+                        pendingReminder.getContent(), pendingReminder.getCreatedAt());
+                pendingReminder.delete();
+            }));
+    }
+
+    private static void sendReminder(@NotNull JDA jda, long id, long guildId, long channelId,
+            long authorId, @NotNull CharSequence content, @NotNull TemporalAccessor createdAt) {
+        RestAction<ReminderRoute> route = computeReminderRoute(jda, guildId, channelId, authorId);
+        sendReminderViaRoute(route, id, content, createdAt);
+    }
+
+    private static RestAction<ReminderRoute> computeReminderRoute(@NotNull JDA jda, long guildId,
+            long channelId, long authorId) {
+        // If guild and channel can still be found, send there
+        Guild guild = jda.getGuildById(guildId);
+        if (guild != null) {
+            TextChannel channel = guild.getTextChannelById(channelId);
+            if (channel != null) {
+                return createGuildReminderRoute(jda, authorId, channel);
+            }
+        }
```

Comment:
```suggestion
        TextChannel channel = jda.getTextChannelById(channelId);
        if (channel != null) {
            return createGuildReminderRoute(jda, authorId, channel);
        }
```

Reply:
Oh, the channel ids are unique across guilds as well? Cool, thanks.